tests/app/data/test_schema_validation.py

```diff
@@ -41,15 +47,67 @@ def test_schema(self):
             self.fail("{} Schema Validation Errors.".format(len(errors)))
 
     @staticmethod
-    def validate_json_against_schema(file, json_file, schema):
+    def validate_json_against_schema(file, json_to_validate, schema):
         try:
-            json_to_validate = json.load(json_file)
             validate(json_to_validate, schema)
         except jsonschema.exceptions.ValidationError as e:
             return "Schema Validation Error! File [{}] does not validate against schema. Error [{}]".format(file, e)
         except JSONDecodeError as e:
             return "JSON Parse Error! Could not parse [{}]. Error [{}]".format(file, e)
 
 
+    @staticmethod
+    def validate_schema_contains_valid_routing_rules(file, json_to_validate):
+        for block in SchemaHelper.get_blocks(json_to_validate):
+            if 'routing_rules' in block and len(block['routing_rules']) > 0:
+                for rule in block['routing_rules']:
+                    is_goto_rule = 'goto' in rule and 'when' in rule['goto'].keys() or 'id' in rule['goto'].keys()
```

Comment:
Do we need this check? `'when' in rule['goto'].keys() or ` Isn't it sufficient just to check for existence of `'id'`?